warehouse/internal/repo/schema.go

```diff
@@ -192,3 +193,56 @@ func (repo *WHSchema) GetNamespace(ctx context.Context, sourceID, destID string)
 
 	return namespace, nil
 }
+
+func (repo *WHSchema) GetTablesForConnection(ctx context.Context, connections []warehouseutils.Connection) ([]warehouseutils.Connection, error) {
+	if len(connections) == 0 {
+		return nil, fmt.Errorf("no source id and destination id pairs provided")
+	}
+
+	// select max id for each source id and destination id pair
+	innerQueryTemplate := `SELECT max(id) FROM ` + whSchemaTableName + `
+		WHERE
+			(source_id, destination_id) IN (%s)
+		AND
+			namespace <> ''
```

Comment:
[Question] why do you ignore empty namespaces ?

Reply:
My idea was that if the namespace was empty the schema column will be empty as well. 
I have changed this condition to a direct check `schema::text <> '{}'::text`

Reply:
namespace will never be empty i think